src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INITIAL_CAPACITY = 16;
+    private Node<K, V>[] nodeArray = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    private int size;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
+        Node<K,V> newNode = new Node<>(key,value,null);
+        int index = hash(key) % nodeArray.length;
+        threshold = (int) (nodeArray.length * LOAD_FACTOR);
+
+        if (size > threshold) {
+            resize();
+        }
+
+        if (nodeArray[index] == null) {
+            nodeArray[index] = newNode;
+            size++;
+            return;
+        }
 
+        for (Node<K,V> node = nodeArray[index]; node != null; node = node.next) {
+            if (isEquals(node, key)) {
+                node.value = value;
+                return;
+            } else if (node.next == null) {
+                node.next = newNode;
+                size++;
+                return;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key) % nodeArray.length;
+        for (Node<K,V> node = nodeArray[index]; node != null; node = node.next) {
+            if (isEquals(node, key)) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> newt) {
+            this.key = key;
+            this.value = value;
+            this.next = newt;
+        }
+
+        public K getKey() {
```

Comment:
**Don’t use getters and setters in class Node**
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/hash-map/hash-map)